### PR TITLE
Dockerfiles for Fedora, Ubuntu and Rocky #1535, #1537, #1538

### PR DESCRIPTION
**Pull request by Jeeppler**
- added the ability to change the ports #1965
- Start scripts for Ubuntu (otherwise same as Debian) #1537
- Dockerfile and Java installer scripts for RockyLinux #1538
- Dockerfile and Java installer scripts for Fedora #1535

The main reason for these additional scripts is compatibility testing.

Each of the distributions is chosen for a particular reason:

- RockyLinux is binary compatible to RedHat Enterprise Linux and others. It is the successor of CentOS (which became a rolling release distribution).
- Fedora is both used on servers and it is incorporating several new features from the overall GNU/Linux ecosystem.
- Ubuntu, because it is quite popular and the basis for several other distributions.

closes: #1535
closes: #1537
closes: #1538

### CODE REVIEW

**Comment by Jeeppler** [comment]
Still waiting for adoptium/installer/issues/611 to be fixed.

File: sechub-solution/docker/SecHub-Fedora.dockerfile [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+# Global Variables
+#-------------------
+
+# The image argument needs to be placed on top
+ARG BASE_IMAGE
+
+# Inject the target architecture
+# For more information: 
+# - https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+ARG TARGETARCH
```
Comment (sven-dmlr) [inline sechub-solution/docker/SecHub-Fedora.dockerfile:13]: Good to know :+1:

File: sechub-solution/docker/SecHub-Fedora.dockerfile [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+# Build args
+ARG BUILD_TYPE="download"
+
+ARG SECHUB_VERSION="0.35.2"
+ARG TAG=""
+ARG BRANCH=""
+
+ARG GO="go1.20.1.linux-${TARGETARCH}.tar.gz"
```
Comment (sven-dmlr) [inline sechub-solution/docker/SecHub-Fedora.dockerfile:22]: This argument is never used. You install the distribution's Golang package.
```suggestion
```

File: sechub-solution/docker/SecHub-Fedora.dockerfile [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ARG SECHUB_ARTIFACT_FOLDER="/artifacts"
+
+#-------------------
+# Builder Build
+#-------------------
+
+FROM ${BASE_IMAGE} AS builder-build
+
+# Build args
+ARG GO
```
Comment (sven-dmlr) [inline sechub-solution/docker/SecHub-Fedora.dockerfile:39]:
```suggestion
```

File: sechub-solution/docker/install-java/fedora/install-java.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    exit 1
+fi
+
+if [ -z "$JAVA_VERSION" ]
+then
+    print_error "ERROR: No Java version provided!"
+    exit 1
+fi
+
+if [ -z "$JAVA_RUNTIME" ]
```
Comment (sven-dmlr) [inline sechub-solution/docker/install-java/fedora/install-java.sh:28]: If the possible values are jre and jdk, you could check this
```suggestion
if [ "$JAVA_RUNTIME" != "jdk" && "$JAVA_RUNTIME" != "jre" ]
```

File: sechub-solution/docker/install-java/rocky/install-java.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    exit 1
+fi
+
+if [ -z "$JAVA_VERSION" ]
+then
+    print_error "ERROR: No Java version provided!"
+    exit 1
+fi
+
+if [ -z "$JAVA_RUNTIME" ]
```
Comment (sven-dmlr) [inline sechub-solution/docker/install-java/rocky/install-java.sh:28]:
```suggestion
if [ "$JAVA_RUNTIME" != "jdk" && "$JAVA_RUNTIME" != "jre" ]
```

**Review by sven-dmlr — CHANGES_REQUESTED** [review]
Good job.

Found some minor things.
Please check my comments.

**Review by sven-dmlr — APPROVED** [review]

**Pull request merged**